tensorflow_io/core/plugins/file_system_plugins.cc

```diff
@@ -25,8 +31,14 @@ void TF_InitPlugin(TF_FilesystemPluginInfo* info) {
   tensorflow::io::az::ProvideFilesystemSupportFor(&info->ops[0], "az");
   tensorflow::io::http::ProvideFilesystemSupportFor(&info->ops[1], "http");
   tensorflow::io::s3::ProvideFilesystemSupportFor(&info->ops[2], "s3e");
-  tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[3], "hdfse");
-  tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[4], "viewfse");
-  tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[5], "hare");
+  if (enable_legacy == "true" || enable_legacy == "1") {
+    tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[3], "hdfse");
+    tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[4], "viewfse");
+    tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[5], "hare");
+  } else {
+    tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[3], "hdfs");
+    tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[4], "viewfs");
+    tensorflow::io::hdfs::ProvideFilesystemSupportFor(&info->ops[5], "har");
+  }
```

Comment:
@yongtang wouldn't the else condition conflict with tensorflow's implementation? even tf would be exposing `hdfs` scheme right?

Reply:
@kvignesh1420 In tensorflow-io, `hdfs` will be enabled if no `TF_ENABLE_LEGACY_FILESYSTEM` provided (`enable_legacy != true`) while in tensorflow `hdfs` will not be enabled. So:
- TF_ENABLE_LEGACY_FILESYSTEM=1: `hdfs` enabled in tensorflow, disabled (switched to `hdfse`) in tensorflow-io.
- TF_ENABLE_LEGACY_FILESYSTEM=0: `hdfs` disabled in tensorflow, enabled in tensorflow-io.

Reply:
Thanks for the confirmation.